src/Streamly/Internal/Data/Unfold/Types.hs

```diff
@@ -63,3 +67,86 @@ map f (Unfold ustep uinject) = Unfold step uinject
 instance Functor m => Functor (Unfold m a) where
     {-# INLINE fmap #-}
     fmap = map
+
+------------------------------------------------------------------------------
+-- Applicative
+------------------------------------------------------------------------------
+
+{-# INLINE const #-}
+const :: Applicative m => m b -> Unfold m a b
+const m = Unfold step inject
+
+    where
+
+    inject _ = pure False
+
+    step False = (`Yield` True) <$> m
+    step True = pure Stop
```

Comment:
This does not feel right to me. Discarding the `inject` value should be signified in the type.

```
const :: Applicative m => m b -> Unfold m Void b
```

Reply:
`const` has to ignore it, you get that from the regular `const :: a -> b -> a`

Reply:
We want to accept an input and discard it. I am not sure how that would be possible with `Void` input type.